astacus/node/sqlite_snapshot.py

```diff
@@ -89,7 +89,14 @@ def _con(self) -> sqlite3.Connection:
             self.db.unlink()
         else:
             self.db.parent.mkdir(parents=True, exist_ok=True)
-        con = sqlite3.connect(self.db, isolation_level=None, check_same_thread=False)
+        # Python 3.12+ fetch result are not consistent in multi-threading application
+        # and causing an API misuse error.
+        # The multithreading use is intended and supported because threadsafety is set to 3.
+        # In serialized mode, SQLite can be safely used by multiple threads with no restriction.
+        # Current solution is to set cached_statements=0 when connecting to the database.
+        # See issue https://github.com/python/cpython/issues/118172
+        assert sqlite3.threadsafety == 3, "sqlite is not multithreading safe."
```

Comment:
nit: IMHO this check should be around Astacus startup somewhere. But could be addressed in a separate PR if re-testing is inconvenient, this can go in as-is

Reply:
> IMHO this check should be around Astacus startup somewhere
IMHO it could also be placed at package build/test time, since we build a RPM for a specific version, it's unlikely that then we run with a different dependency set which is suddenly changing at runtime. We should have a test with that specific check.